leku/src/main/res/layout/toolbar_search.xml

```diff
@@ -1,49 +1,51 @@
-<android.support.v7.widget.Toolbar
+<android.support.design.widget.AppBarLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:id="@+id/map_search_toolbar"
-    android:layout_height="?attr/actionBarSize"
+    android:layout_height="wrap_content"
     android:layout_width="match_parent"
-    android:layout_marginTop="@dimen/gap"
-    android:layout_marginLeft="@dimen/gap"
-    android:layout_marginRight="@dimen/gap"
-    android:elevation="2dp"
-    android:textAppearance="@color/abc_primary_text_material_dark"
-    android:background="@drawable/toolbar_custom_background"
-    app:theme="@style/ThemeOverlay.AppCompat.Light"
-    app:popupTheme="@style/ThemeOverlay.AppCompat.Light"
+    android:padding="@dimen/gap"
+    app:elevation="@dimen/appbar_search_elevation"
+    android:background="@android:color/transparent"
     >
-
-  <LinearLayout
-      android:id="@+id/leku_search_touch_zone"
+  <android.support.v7.widget.Toolbar
+      android:id="@+id/map_search_toolbar"
+      android:layout_height="?actionBarSize"
       android:layout_width="match_parent"
-      android:layout_height="wrap_content"
-      android:layout_gravity="center_vertical"
-      android:orientation="horizontal"
+      android:background="@drawable/toolbar_custom_background"
+      app:theme="@style/ThemeOverlay.AppCompat.Light"
+      app:popupTheme="@style/ThemeOverlay.AppCompat.Light"
       >
 
-    <EditText
-        android:id="@+id/leku_search"
+    <LinearLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:inputType="text"
-        android:imeOptions="actionSearch"
-        android:background="@color/search_background"
-        android:hint="@string/search_hint"
-        android:layout_weight="1"
         android:layout_gravity="center_vertical"
-        />
+        android:orientation="horizontal"
+        >
 
-    <ImageView
-        android:id="@+id/leku_clear_search_image"
-        android:layout_width="24dp"
-        android:layout_height="wrap_content"
-        android:src="@drawable/ic_clean"
-        android:tint="@color/strong_grey"
-        android:contentDescription="@string/search_hint"
-        android:layout_weight="0"
-        android:visibility="invisible"
-        />
-  </LinearLayout>
-</android.support.v7.widget.Toolbar>
\ No newline at end of file
+      <EditText
+          android:id="@+id/leku_search"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:inputType="text"
+          android:imeOptions="actionSearch"
+          android:background="@color/search_background"
+          android:hint="@string/search_hint"
+          android:layout_weight="1"
+          android:layout_gravity="center_vertical"
+          />
+
+      <ImageView
+          android:id="@+id/leku_clear_search_image"
+          android:layout_width="24dp"
```

Comment:
What do you think about extracting `24dp` to `dimens.xml`? It will make easier to catch if this value is a tailored value for this view, or something reused in the whole project.